docs/InputReferenceGuide.rst

```diff
@@ -0,0 +1,157 @@
+| OpExecutionMode    | .. code-block:: llvm                                    |
+|                    |                                                         |
+|                    |    !spirv.ExecutionMode = !{!0}                         |
+|                    |    !6 = !{void ()* @worker, i32 30, i32 262149}         |
+|                    |    ; Set execution mode with id 30 (VecTypeHint) and    |
+|                    |    ; literal `262149` operand.                          |
++--------------------+---------------------------------------------------------+
+| OpCapability       | .. code-block:: llvm                                    |
+|                    |                                                         |
+|                    |    !spirv.Capability = !{!0}                            |
+|                    |    !0 = !{i32 10} ; Float64 - program uses doubles      |
++--------------------+---------------------------------------------------------+
+| Generator's magic  | .. code-block:: llvm                                    |
+| number - word # 2  |                                                         |
+| in SPIR-V module   |    !spirv.Generator = !{!0}                             |
+|                    |    !0 = !{i16 6, i16 123}                               |
+|                    |    ; 6 - Generator Id, 123 - Generator Version          |
++--------------------+---------------------------------------------------------+
+
+**TODO**: Should we document all OpExecutionMode operands supported by the
+translator? There is a number of non-standard modes added as Intel extensions.
+
+Function level requirements
+===========================
+
+Calling convention
+------------------
+
+OpEntryPoint information is represented in SPIR IR in calling convention.
+A function with `spir_kernel` calling convention will be translated as an entry
+point to SPIR-V module.
+
+Function metadata
+-----------------
+
+Some kernel parameter information is stored in SPIR IR as a function level
+metadata.
+
+For example:
+
+- !kernel_arg_addr_space !1
+- !kernel_arg_access_qual !2
+- !kernel_arg_type !3
+- !kernel_arg_base_type !4
+- !kernel_arg_type_qual !5
+
+**TODO**: Are all these metadata optional? Access qualifiers are translated for
+image types, but they should be encoded in LLVM IR type name rather than
+function metadata.
+
+Address spaces
+==============
+
+Following
+`SPIR-V storage classes <https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#Storage_Class>`_
+are naturally represented as LLVM IR address spaces with following mapping:
+
+SPIR-V storage class    LLVM IR address space
+Function                No address space
+CrossWorkgroup          addrspace(1)
+UniformConstant         addrspace(2)
+Workgroup               addrspace(3)
+Generic                 addrspace(4)
+
+Unsupported LLVM IR features
+============================
+
+**TODO**: Should these features be listed explicitly?
```

Comment:
It might be easier to list supported features, because it is smaller and it is not updated so often as list of unsupported ones (LLVM development moves faster than the translator development)